packages/icons/src/PentahoIconWrapper.tsx

```diff
@@ -0,0 +1,90 @@
+import styled from "@emotion/styled";
+import { SizeProp } from "@fortawesome/fontawesome-svg-core";
+import {
+  FontAwesomeIcon,
+  FontAwesomeIconProps,
+} from "@fortawesome/react-fontawesome";
+import { getColor, HvColorAny, HvSize } from "@hitachivantara/uikit-styles";
+
+import { IconSize, svgSizeMap } from "./IconBase";
+
+export interface PentahoIconWrapperProps
+  extends Omit<FontAwesomeIconProps, "color" | "size"> {
+  size?: HvSize;
+  iconSize?: IconSize;
+  color?: HvColorAny;
+  container?: boolean;
+}
+
+const getContainerSize = (iconSize?: IconSize | HvSize) => {
+  if (!iconSize) return;
+  const baseSize = svgSizeMap[iconSize] || svgSizeMap.sm;
+  // container has 8px margin on each side, except for XS which has 10px
+  const containerSize =
+    baseSize + 2 * (iconSize === "XS" || iconSize === "xs" ? 10 : 8);
+
+  return { width: containerSize, height: containerSize };
+};
+
+const getSVGSize = (iconSize?: IconSize | HvSize) => {
+  if (!iconSize) return;
+  const baseSize = svgSizeMap[iconSize] || svgSizeMap.sm;
+  return { width: baseSize, height: baseSize };
+};
+
+const StyledIconContainer = styled("div")({
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+});
+
+export const PentahoIconWrapper = ({
+  icon,
+  color: colorProp,
+  iconSize,
+  size,
+  container = true,
+  fontSize,
+  ...others
+}: PentahoIconWrapperProps) => {
+  const getSVGsizeTranslation = (
+    iconSize?: IconSize | HvSize,
+    size?: HvSize,
+  ): SizeProp | undefined => {
+    if (iconSize) {
+      switch (iconSize) {
+        case "XS" || "xs":
+          return "xs";
+        case "S" || "sm":
+          return "1x";
+        case "L" || "lg":
+          return "6x";
+        default: // "M"
+          return "2x";
+      }
+    }
+  };
+
+  const iconComponent = (
+    <FontAwesomeIcon
+      icon={icon}
+      size={getSVGsizeTranslation(iconSize, size)}
+      fontSize={iconSize ? 16 : fontSize}
+      style={getSVGSize(iconSize ?? size)}
+      color={colorProp ? getColor(colorProp) : "inherit"}
```

Comment:
`getColor` deals with fallback values

```suggestion
      color={getColor(colorProp, "inherit")}
```